components/base/wheeled/wheeled_base.go

```diff
@@ -396,3 +403,78 @@ func CreateWheeledBase(
 
 	return base, nil
 }
+
+type kinematicWheeledBase struct {
+	*wheeledBase
+	slam  slam.Service
+	model referenceframe.Model
+}
+
+func (base *wheeledBase) WrapWithKinematics(slam slam.Service) (base.KinematicBase, error) {
+	var err error
+	kwb := &kinematicWheeledBase{
+		wheeledBase: base,
+		slam:        slam,
+	}
+	kwb.model, err = kwb.buildModel(base.cfg)
+	return kwb, err
+}
+
+func (kwb *kinematicWheeledBase) ModelFrame() referenceframe.Model {
+	return kwb.model
+}
+
+func (kwb *kinematicWheeledBase) CurrentInputs(ctx context.Context) ([]referenceframe.Input, error) {
+	// TODO(RSDK-2311): complete the implementation
+	return []referenceframe.Input{}, errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) GoToInputs(ctx context.Context, goal []referenceframe.Input) error {
+	// TODO(RSDK-2311): complete the implementation
+	return errors.New("not implemented yet")
+}
+
+func (kwb *kinematicWheeledBase) buildModel(cfg config.Component) (referenceframe.Model, error) {
+	// TODO(rb): this is a hacky workaround for not having kinematics for bases yet
```

Comment:
q: how are we going to deal with components whose parent is the base and have their own geometry?

Reply:
That is future work.  The whole idea was to remove kinematics from the equation for now, so that we don't have that added complexity. Because of that, this code is not very good and should in the near term be replaced with something more robust.

Reply:
I'm totally fine with not implementing that kinematics right now (as was scoped).

When I say "deal with" I'm thinking more like, erroring if the base has children with geometries, or at least, spouting off a bunch of loud errors into the log.

Reply:
Yeah thats a good question.  

My thought is this would have to happen from the motion service itself, since that has access to the whole frame system. We could bake some logic into that when you go to make a call to specifically move a base.  Since we don't have access to the rest of the frame system here, I think this is out of scope for this particular PR, and should be something that we pick up when we move to do E

Reply:
I'm good with that approach.